testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -86,6 +90,8 @@
     private final AutoCloseableCloser afterClassCloser = AutoCloseableCloser.create();
     private io.trino.sql.query.QueryAssertions queryAssertions;
 
+    private final ThreadLocal<Optional<CallMutator>> callMutatorThreadLocal = ThreadLocal.withInitial(Optional::empty);
```

Comment:
Why a thread local? Thread locals are generally a code smell (passing arguments via a side channel, etc)

Reply:
Agree. Actually after discussion with @findepi we decided we do not need last commit of this change at all. Will update PR in a while.